sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -447,7 +447,11 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       // Create the table if the table didn't exist.
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
-        val sql = s"CREATE TABLE $table ($schema)"
+        // To allow certain options to append when create a new table, which can be
+        // table_options or partition_options.
+        // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        val createtblOptions = extraOptions.getOrElse("createTableOptions", "")
```

Comment:
A quick question here. The JDBCOptions also contain URL and table information, like:
```
 // a JDBC URL
  val url = parameters.getOrElse("url", sys.error("Option 'url' not specified"))
  // name of table
  val table = parameters.getOrElse("dbtable", sys.error("Option 'dbtable' not specified"))
```
Shall we merge those information into current DataFrameWriter as well?